derive/src/lib.rs

```diff
@@ -119,3 +120,16 @@ pub fn pypayload(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
     result_to_tokens(pypayload::impl_pypayload(input))
 }
+
+/// use on struct with named fields like `struct A{x:i32, y:i32}` to impl `Trace` for datatype
+///
+/// use `#[notrace]` on fields you wish not to trace
+#[proc_macro_attribute]
```

Comment:
> This could probably be a normal derive() macro, and then you wouldn't have to deal with removing the notrace attr

Yes, `#[notrace]` is only used four time in th following, still considering whether or not to remove it and hand write Trace:
```
#[pyclass(module = false, name = "enumerate")]
#[derive(Debug)]
#[pytrace]
pub struct PyEnumerate {
    #[notrace]
    counter: PyRwLock<BigInt>,
    iterator: PyIter,
}
```
```
#[pyclass(module = false, name = "iterator")]
#[derive(Debug)]
#[pytrace]
pub struct PySequenceIterator {
    // cached sequence methods
    #[notrace]
    seq_methods: &'static PySequenceMethods,
    internal: PyMutex<PositionIterInternal<PyObjectRef>>,
}
```

```
#[pyclass(module = false, name = "zip")]
#[derive(Debug)]
#[pytrace]
pub struct PyZip {
    iterators: Vec<PyIter>,
    #[notrace]
    strict: PyAtomic<bool>,
}
```

```
#[derive(Debug, Clone)]
#[pytrace]
pub struct ArgMapping {
    obj: PyObjectRef,
    #[notrace]
    methods: &'static PyMappingMethods,
}
```